pkg/runtime/framework/plugins/jobset/builder.go

```diff
@@ -115,17 +116,23 @@ func (b *Builder) Launcher() *Builder {
 // Trainer updates JobSet values for the trainer Job.
 func (b *Builder) Trainer(info *runtime.Info, trainJob *trainer.TrainJob) *Builder {
 	for i, rJob := range b.Spec.ReplicatedJobs {
-		if *rJob.Name == constants.JobTrainerNode {
+		jobMetadata := rJob.Template.ObjectMetaApplyConfiguration
+		if jobMetadata == nil || jobMetadata.Labels == nil {
+			continue
+		}
+		if ancestor, ok := jobMetadata.Labels[constants.LabelTrainJobAncestor]; ok && ancestor == constants.AncestorTrainer {
 			// TODO: Support multiple replicas ('.template.spec.replicatedJobs[*].replicas') for replicated Jobs.
 			// REF: https://github.com/kubeflow/trainer/issues/2318
 			b.Spec.ReplicatedJobs[i].Replicas = ptr.To[int32](1)
 			// Update the Parallelism and Completions values for the Trainer Job.
-			b.Spec.ReplicatedJobs[i].Template.Spec.Parallelism = info.TemplateSpec.PodSets[i].Count
-			b.Spec.ReplicatedJobs[i].Template.Spec.Completions = info.TemplateSpec.PodSets[i].Count
+			b.Spec.ReplicatedJobs[i].Template.Spec.Parallelism = info.FindPodSetByAncestor(constants.AncestorTrainer).Count
+			b.Spec.ReplicatedJobs[i].Template.Spec.Completions = info.FindPodSetByAncestor(constants.AncestorTrainer).Count
 
 			// Update values for the Trainer container.
 			for j, container := range rJob.Template.Spec.Template.Spec.Containers {
-				if *container.Name == constants.ContainerTrainer {
+				// TODO: This `Launcher` should be handled more correctly.
+				// Currently, we assume the 'launcher' container name is used for MPI Launcher.
```

Comment:
Can we always use `node` container name for the launcher ReplicatedJob ?

Reply:
Using same container or initContainer name is not allowed in Kubernetes, which means Pod creation failed

Reply:
Yes, removal initContainer implementations allow us to remove this

Reply:
Done.